src/modules/samples/rust/src/sample.rs

```diff
@@ -181,7 +181,8 @@ impl Sample {
             if self.max_payload_size_bytes != 0
             && payload.len() as u32 > self.max_payload_size_bytes {
                 // In other modules you could prioritize parts of the payload
-                Err(MmiError::InvalidArgument)
+                println!("Payload size exceeded max payload size bytes in get");
+                Err(MmiError::PayloadSizeExceeded)
```

Comment:
For MmiGet, exceeding the maximum size cannot result in an error. For MmiGet the module needs to truncate reported payload so that it fits the maximum size indicated by MmiOpen.

Please fix this.

Failing MmiSet for payload that exceeds maximum size is ok.

Reply:
Changed to truncate the payload although this branch should never trigger based on the behavior of the sample module. It saves in the object whatever was passed into set. And the set payload must be below max_payload_size_bytes, so when the stored value is reconverted into a string payload, it will still be below max_payload_size_bytes.